cdm/src/main/java/ucar/nc2/ft2/coverage/simpgeometry/CFLine.java

```diff
@@ -0,0 +1,273 @@
+		this.points.clear();
+		Array xPts = null;
+		Array yPts = null;
+		Variable node_counts = null;
+		Variable part_node_counts = null;
+
+		List<CoordinateAxis> axes = dataset.getCoordinateAxes();
+		CoordinateAxis x = null; CoordinateAxis y = null;
+		
+		String[] node_coords = var.findAttributeIgnoreCase(CF.NODE_COORDINATES).getStringValue().split(" ");
+		
+		// Look for x and y
+		
+		for(CoordinateAxis ax : axes){
+			
+			if(ax.getFullName().equals(node_coords[0])) x = ax;
+			if(ax.getFullName().equals(node_coords[1])) y = ax;
+		}
+		
+		// Affirm node counts
+		String node_c_str = var.findAttValueIgnoreCase(CF.NODE_COUNT, "");
+		
+		if(!node_c_str.equals("")) {
+			node_counts = dataset.findVariable(node_c_str);
+		}
+		
+		else return null;
+		
+		// Affirm part node counts
+		String p_node_c_str = var.findAttValueIgnoreCase(CF.PART_NODE_COUNT, "");
+		
+		if(!p_node_c_str.equals("")) {
+			part_node_counts = dataset.findVariable(p_node_c_str);
+		}
+		
+		SimpleGeometryIndexFinder indexFinder = new SimpleGeometryIndexFinder(node_counts);
+		
+		//Get beginning and ending indicies for this polygon
+		int lower = indexFinder.getBeginning(index);
+		int upper = indexFinder.getEnd(index);
+
+		
+		try {
+			
+			xPts = x.read( lower + ":" + upper ).reduce();
+			yPts = y.read( lower + ":" + upper ).reduce(); 
+
+			IndexIterator itr_x = xPts.getIndexIterator();
+			IndexIterator itr_y = yPts.getIndexIterator();
+			
+			// No multipolygons just read in the whole thing
+			if(part_node_counts == null) {
+				
+				this.next = null;
+				this.prev = null;
+				
+				// x and y should have the same shape, will add some handling on this
+				while(itr_x.hasNext()) {
+					this.addPoint(itr_x.getDoubleNext(), itr_y.getDoubleNext());
+				}
+	
+				this.setData(var.read(":," + index).reduce());
+			}
+			
+			// If there are multipolygons then take the upper and lower of it and divy it up
+			else {
+				
+				CFLine tail = this;
+				Array pnc = part_node_counts.read();
+				IndexIterator pnc_itr = pnc.getIndexIterator();
+				
+				// In part node count search for the right index to begin looking for "part node counts"
+				int pnc_ind = 0;
+				int pnc_end = 0;
+				while(pnc_end < lower)
+				{
+					pnc_end += pnc_itr.getIntNext();
+					pnc_ind++;
+				}
+				
+				// Now the index is found, use part node count and the index to find each part node count of each individual part
+				while(lower < upper) {
+					
+					int smaller = pnc.getInt(pnc_ind);
+					
+					while(smaller > 0) {
+						tail.addPoint(itr_x.getDoubleNext(), itr_y.getDoubleNext());
+						smaller--;
+					}
+					
+					// Set data of each
+					tail.setData(var.read(":," + index));
+					lower += tail.getPoints().size();
+					pnc_ind++;
+					tail.setNext(new CFLine());
+					tail = tail.getNext();
+				}
+				
+				//Clean up
+				tail = tail.getPrev();
+				if(tail != null) tail.setNext(null);
+			}
+		}
+		
+		catch (IOException e) {
+			e.printStackTrace();

```

Comment:
`exception.printStackTrace()` is a handy debugging tool on simpler projects that don't rely on logging tools. On big projects like THREDDS, the quantity of log messages gets so overwhelming that formal logging tools are needed to address the flood of information.  This permits people inspecting the logs and configuring the app to choose what level of detail they want, and to send different kinds of information to different log files. Would you be able to help users of THREDDS better scope this troubleshooting infromation by logging these Exceptions instead of using `e.printStackTrace()`?

DZone has an article on this:
https://dzone.com/articles/java-top-5-exception-handling

I think THREDDS uses SLF4J. I would encourage you to check out other THREDDS classes usages of this logging framework.

Reply:
I was aware of SLF4J being used in THREDDS, however, it looks like there's some spottiness in it being included in NetCDF-All. Will look into it, thanks!

Reply:
A new ticket shall be made for that!